htsget-search/src/htsget/bam_search.rs

```diff
@@ -29,24 +28,14 @@ pub(crate) struct BamSearch<S> {
   storage: Arc<S>,
 }
 
-#[async_trait]
-impl<ReaderType> BlockPosition for AsyncReader<ReaderType>
-where
-  ReaderType: AsyncRead + AsyncSeek + Unpin + Send + Sync,
-{
-  async fn read_bytes(&mut self) -> Option<usize> {
+impl BinningIndexExt for Index {
+  fn get_all_chunks(&self) -> Vec<&Chunk> {
     self
-      .read_record(&mut sam::alignment::Record::default())
-      .await
-      .ok()
-  }
-
-  async fn seek_vpos(&mut self, pos: VirtualPosition) -> io::Result<VirtualPosition> {
-    self.seek(pos).await
-  }
-
-  fn virtual_position(&self) -> VirtualPosition {
-    self.virtual_position()
+      .reference_sequences()
+      .iter()
+      .flat_map(|ref_seq| ref_seq.bins())
+      .flat_map(|bin| bin.chunks())
+      .collect()
```

Comment:
What is the average size in memory for this? I.e, what's the size of a 500GB BAM gzi index?

Thinking about optimal memsize for lambdas.

Reply:
Unfortunately GZI can be somewhat large. The largest BGZF block size if 64kb, so an index entry is required at least every 64kb. A 500GB file will have a GZI file that is at least $((500 GB / 64 KB) * 8 * 2)  \approx 125 MB$ big.

However, most of the data in the in the GZI is not used by `Search`, as it only requires the next virtual position after a queried chunk. This means that the data required in the GZI by `Search` shouldn't be any larger than the regular index file (e.g. BAI). It would be fairly easy to implement a tool that removes unnecessary entries from GZI that htsget-rs doesn't need. Also, these resulting GZI files should (I think) be valid GZI files, albeit with a lot of entries missing. There's also a lot of repeated and unnecessary data in the regular index files too, which htsget-rs has no use for.